ipaserver/install/server/upgrade.py

```diff
@@ -383,31 +384,42 @@ def ca_enable_ldap_profile_subsystem(ca):
         logger.info('CA is not configured')
         return False
 
+    instance = pki.server.instance.PKIInstance('pki-tomcat')
+    instance.load()
+
+    subsystem = instance.get_subsystem('ca')
+
+    # find profile subsystem in CA configuration file
+    n = None
+    pattern = re.compile(r'subsystem\.(\d+)\.id')
+    for name in subsystem.config:
+        match = pattern.match(name)
+        if not match:
+            continue
+
+        value = subsystem.config[value]
+        if value != 'profile':
+            continue
+
+        n = match.group(1)
+        break
+
+    if not n:
+        raise RuntimeError(
+            'Unable to find profile subsystem in %s' % paths.CA_CS_CFG_PATH)
+
+    # get profile subsystem class name
+    class_name = subsystem.config['subsystem.%s.class' % n]
+
     needs_update = False
-    directive = None
-    try:
-        for i in range(15):
-            directive = "subsystem.{}.class".format(i)
-            value = directivesetter.get_directive(
-                paths.CA_CS_CFG_PATH,
-                directive,
-                separator='=')
-            if value == 'com.netscape.cmscore.profile.ProfileSubsystem':
```

Comment:
This can be accomplished with a much smaller change - a one-liner in fact, and it will not depend on a newer version of pki-core.  Just change this comparision from
```python
             if value == 'com.netscape.cmscore.profile.ProfileSubsystem':
```
to
```python
             if value != 'com.netscape.cmscore.profile.LDAPProfileSubsystem':
```

Is there any reason not to keep it simple like this?

Reply:
That might only work if the profile subsystem is the first one in the list. If the first subsystem is not a profile subsystem, the code will set `needs_update` to `True` immediately and break from the loop.

I think we need to make sure that `subsystem.<n>.id=profile` first before comparing the class name. That probably can be done without importing `pki.server` module. What do you think?

Reply:
Also the code imposes an artificial limit on the number of subsystems it's checking:
```
for i in range(15):
```
While it works now, it could break under certain circumstances.

Reply:
Of course, you are right.  But just add a couple more lines to first check if  `subsystem.<n>.id == 'profile'`.  So it is closer to a 10 line patch than a 1-liner :)

I accept your point about the limit of 15.  Well, maybe we just increase it (to 100, 1000, whatever), and ensure that we break after we have encountered the profile subsystem configuration.  I accept it is not perfect but IMO the risk is so low it hardly seems worth a more complicated patch that depends on unreleased Dogtag version.